Fix TransactionTooLargeException when opening HistoryDetailActivity

RenanLukas opened the pull request:
Fixes #16012

**Depends on: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2310**

To test:
1 - Log in
2 - Select a site
3 - On "My Site", tap on "Posts"
4 - Tap on a post that was updated at least once
5 - Tap on overflow menu (three vertical dots on top right)
6 - Tap on "History"
7 - Tap on any Revision on list. Revision details screen should open, instead of the app crashing.

## Regression Notes
1. Potential unintended areas of impact
Revision details ViewPager (introduced a bug while developing and then fixed it)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing opening history details screen with different sites, moving the ViewPager from beginning to end.

3. What automated tests I added (or what prevented me from doing so)
I've tried to update `HistoryViewModelTest` with my changes, but couldn't do it because `OnRevisionsFetched` visibility is limited. If there are any ideas to have this code tested the way it is, please let me know.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

RenanLukas committed:
Replace Revision list parameter in HistoryDetailActivity with Revision IDs to avoid TransactionTooLargeException (the new parameters are used to fetch the revisions)

RenanLukas commented:
👀 CI

RenanLukas committed:
Suppress detekt: TooManyFunctions for HistoryViewModel

RenanLukas committed:
Update FluxC version to updated to the version that contains PostStore.getRevisionById

khaykov commented on WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:
```diff
@@ -1941,8 +1941,20 @@ public void onBackPressed() {
+    }
 
-        ActivityLauncher.viewHistoryDetailForResult(this, mRevision, revisions);
+    private long[] getRevisionsIds(@NonNull final List<Revision> revisions) {
```
If I understand this correctly, number of returned ID's will be same as the number of revisions, so maybe we don't need to extract them into `idsList` first?

Can we simplify it into something like this (not tested)?:

```
  private long[] getRevisionsIds(@NonNull final List<Revision> revisions) {
        final long[] idsArray = new long[revisions.size()];
        for (int i = 0; i < revisions.size(); i++) {
            final Revision current = revisions.get(i);
            idsArray[i] = current.getRevisionId();
        }
        return idsArray;
    }
```

khaykov commented on WordPress/src/main/java/org/wordpress/android/ui/history/HistoryDetailContainerFragment.java:
```diff
@@ -172,6 +179,37 @@ public void onClick(View view) {
+        final ArrayList<Revision> revisions = new ArrayList<>();
+        for (int i = 0; i < revisionModels.size(); i++) {
+            final RevisionModel current = revisionModels.get(i);
+            final Revision revision = new Revision(current.getRevisionId(), current.getDiffFromVersion(),
```
`Revision` has constructor that takes a `RevisionModel`. Wonder if we can utilize it?

khaykov commented on WordPress/src/main/java/org/wordpress/android/viewmodel/history/HistoryViewModel.kt:
```diff
@@ -223,6 +223,18 @@ class HistoryViewModel @Inject constructor(
+
+    private fun removeRevisionsFromLocalDB(post: PostModel, revisions: List<RevisionModel>) {
+        revisions.forEach {
+            postStore.deleteLocalRevision(it, site, post)
```
If we only remove the revisions received from the endpoint, there is a chance we might have some revisions not removed, if for some reason they are not in the response. What do you think about removing all the revision for specific post?

khaykov commented on WordPress/src/main/java/org/wordpress/android/ui/history/HistoryDetailContainerFragment.java:
```diff
@@ -172,6 +179,37 @@ public void onClick(View view) {
+            final long[] previousRevisionsIds = getArguments().getLongArray(EXTRA_PREVIOUS_REVISIONS_IDS);
+            final List<RevisionModel> revisionModels = new ArrayList<>();
+            for (final long revisionId : previousRevisionsIds) {
+                revisionModels.add(mPostStore.getRevisionById(revisionId));
```
This is probably not a significant issue, but I wonder if performance might suffer when accessing DB from main thread for a post wit a lot of changes.

khaykov reviewed (commented):
Thanks for tackling this one, @renanferrari 🙇  It works well! I left couple of comments in the code.

RenanLukas committed:
Apply PR suggestion: pass site ID and post ID to get revision by ID

RenanLukas committed:
Apply PR suggestion: remove all reviews from specific Post in HistoryViewModel instead of just the ones returned by remote

RenanLukas committed:
Apply PR suggestion: use Revision constructor that receives a RevisionModel instead of passing every field

RenanLukas committed:
Apply PR suggestion: simplify map revision IDs code in EditPostActivity

RenanLukas committed:
Fix checkstyle in ActivityLauncher: missing space

RenanLukas committed:
Merge branch 'trunk' into issue/16012-exception-opening-post-revisions

RenanLukas committed:
Update fluxC version

RenanLukas committed:
Apply PR suggestion: fix indentaion in HistoryViewModel

RenanLukas committed:
Update fluxC version

RenanLukas commented:
Thanks for the review @khaykov ! I've addressed the issues we discussed 👍  (except [this one](https://github.com/wordpress-mobile/WordPress-Android/pull/16105#discussion_r827331041) - if you think it's worth, I'll create a separate issue for it and will work on it in the future)

khaykov reviewed (approved):
Perfect, thank you! lgtm 🚀
